cpp/src/decisiontree/decisiontree_impl.cuh

```diff
@@ -320,11 +326,8 @@ void DecisionTreeBase<T, L>::plant(
       CUML_LOG_WARN("Using experimental backend for growing trees\n");
     }
     T *quantiles = tempmem->d_quantile->data();
-    int *colids = (int *)tempmem->device_allocator->allocate(
-      sizeof(int) * ncols, tempmem->stream);
-    MLCommon::iota(colids, 0, 1, ncols, tempmem->stream);
```

Comment:
Nice to see `colids` being removed. Device-side sampling is indeed the way to go.